File: my.go
package main

import (
	"fmt"
)

var arraySize = 10

type MyStruct struct {
	Roll  int
	Name  string
	Marks float32
}

func main() {
	array := myFunction1()
	fmt.Println(array)

	arrayOfPointers := myFunction2()
	fmt.Println(arrayOfPointers)
}

func myFunction1() []*MyStruct {
	array := make([]*MyStruct, arraySize)
	for i := 0; i < arraySize; i++ {
		array[i] = &MyStruct{
			Roll:  i + 1,
			Name:  fmt.Sprintf("Student %d", i+1),
			Marks: float32(i) * 10,
		}
	}
	return array
}

func myFunction2() *[]MyStruct {
	array := make([]MyStruct, arraySize)
	for i := 0; i < arraySize; i++ {
		array[i] = MyStruct{
			Roll:  i + 1,
			Name:  fmt.Sprintf("Student %d", i+1),
			Marks: float32(i) * 10,
		}
	}
	return &array
}

File: my.c
#include <stdio.h>


int arraySize = 10;

int main()
{
    struct MyStruct *array = myFunction();
    
    struct MyStruct **array = myFunction();

    

    return 0;
}

struct MyStruct
{
    int roll;
    char name[20];
    float marks;
};

struct MyStruct *myFunction()
{
    struct MyStruct *array = malloc(sizeof(struct MyStruct) * arraySize);
    // Initialize array elements

    return array;
}

struct MyStruct **myFunction()
{
    struct MyStruct **array = malloc(sizeof(struct MyStruct *) * arraySize);
    for (int i = 0; i < arraySize; i++)
    {
        array[i] = malloc(sizeof(struct MyStruct));
        // Initialize individual structs
    }
    return array;
}